docs/hazmat/primitives/asymmetric/ec.rst

```diff
@@ -6,34 +6,33 @@ Elliptic curve cryptography
 .. module:: cryptography.hazmat.primitives.asymmetric.ec
 
 
-.. function:: generate_private_key(curve, backend=None)
+.. function:: generate_private_key(curve)
 
     .. versionadded:: 0.5
 
-    Generate a new private key on ``curve`` for use with ``backend``.
+    .. versionchanged:: 36.0
```

Comment:
Is putting one of these on every single function useful?

Reply:
I think that's worth some discussion, which is why I stopped at this PR 😄 

Let me give the primary case: we know that users search and use whatever docs pop up via the search engine. By putting a note on (many) functions noting that the signature has changed we can improve the experience for those people by directing them to the proper docs and lowering confusion. The cost is a bit of noise in the docs for people who are less confused, but this is a big change to people who have expected to pass `backend` or `default_backend()` for the past 8 years (even though we made it optional over a year ago).

Reply:
My arguments against documenting it everywhere:

- It's very noisy, and this may lead to it getting ignored even by people who need it!
- The most useful place for this information is the changelog
- It's annoying to write/review each of these

I could go either way though, truth be told that 3rd one is most motivating

Reply:
If we choose not to document it everywhere, should we still add `.. versionchanged:: 36.0` and just rely on the changelog? Or should we keep the FAQ entry but not link it?

Reply:
Keep FAQ, no vesion changed marker on all functions is my idea.